File: Assets/Game/Scripts/Scene/BaseSceneObject.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class BaseSceneObject
{
    protected int m_Id;
    protected Vector3 m_Position;

    public BaseSceneObject(int id,Vector3 position)
    {
        m_Id = id;
        m_Position = position;
    }

    public int GetId()
    {
        return m_Id;
    }

    public Vector3 GetPosition()
    {
        return m_Position;
    }
}

File: Assets/Game/Lua/General/SceneObject/BaseSceneObject.lua
BaseSceneObject = class()

function BaseSceneObject:ctor(id, position)
    self.pId = id
    self.pPosition = position
end

function BaseSceneObject:GetId()
    return self.pId
end

function BaseSceneObject:GetPosition()
    return self.pPosition
end

function BaseSceneObject:DebugPosition()
    print("对象("..self.pId..")当前位置 x:"..self.pPosition.x.." y:"..self.pPosition.y.." z:"..self.pPosition.z)
end